lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -1,14 +1,51 @@
+# Copyright 2021 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
 import fnmatch
 import ufe
 import maya.mel as mel
 import maya.cmds as cmds
-import mayaUsd.ufe
+import mayaUsd.ufe as mayaUsdUfe
+import mayaUsd.lib as mayaUsdLib
 import maya.internal.common.ufe_ae.template as ufeAeTemplate
+import maya.internal.common.ae.custom as aecustom
+from maya.common.ui import LayoutManager
 
 # We manually import all the classes which have a 'GetSchemaAttributeNames'
 # method so we have access to it and the 'pythonClass' method.
-from pxr import Usd, UsdGeom, UsdLux, UsdRender, UsdRi, UsdShade, UsdSkel, UsdUI, UsdVol
+from pxr import Usd, UsdGeom, UsdLux, UsdRender, UsdRi, UsdShade, UsdSkel, UsdUI, UsdVol, Kind, Tf
+
+def getPrettyName(name):
+    # Put a space in the name when preceded by a capital letter.
+    # Exceptions: Number followed by capital
+    #             Multiple capital letters together
+    prettyName = str(name[0])
+    nbChars = len(name)
+    for i in range(1, nbChars):
+        if name[i].isupper() and not name[i-1].isdigit():
+            if (i < (nbChars-1)) and not name[i+1].isupper():
+                prettyName += ' '
+            prettyName += name[i]
+        elif name[i] == '_':
+            continue
+        else:
+            prettyName += name[i]
+        print('prettyName: "%s"' % prettyName)
 
+    # Make each word start with an uppercase.
+    return prettyName.title()
```

Comment:
This code was down below being used for the schema section names. Its the same rules. So I just created a helper out of it, so I can also use it for the metadata key label.